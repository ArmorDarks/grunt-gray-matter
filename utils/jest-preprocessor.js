/*
Enable testing with coffescript files & React
*/

const coffee = require('coffee-script')

module.exports = {
  process(src, path) {
    if (path.endsWith('.coffee')) {
      return coffee.compile(src, { bare: true })
    }

    return src
  }
}