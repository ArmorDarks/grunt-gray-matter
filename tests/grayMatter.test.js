import { run, grunt } from '../utils/grunt'

const { config, file: { readJSON } } = grunt

describe('Gray Matter task', () => {
  beforeAll(() => run())

  it('should produce correct matter file with defaults for single page', () => {
    expect(readJSON(config('path.build.singleWithDefaults'))).toMatchSnapshot()
  })

  it('should produce correct matter file with defaults for multiple pages', () => {
    expect(readJSON(config('path.build.multipleWithDefaults'))).toMatchSnapshot()
  })

  it('should produce correct matter file with defaults for multiple nested pages', () => {
    expect(readJSON(config('path.build.nestedWithDefaults'))).toMatchSnapshot()
  })
})