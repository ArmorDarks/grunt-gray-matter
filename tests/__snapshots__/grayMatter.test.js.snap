exports[`Gray Matter task should produce correct matter file with defaults for multiple nested pages 1`] = `
Object {
  "tests/fixtures/nested/level2/level3/page4": Object {
    "html": Object {
      "number": 123,
      "test": "Test value",
      "title": "Page4",
      "withDoubleQuotes": "Value with quotes",
      "withQuotes": "Value with quotes",
      "withQuotesInside": "Value\"with\'qoutes\'ins\"ide",
      "withSpaces": "Spaced value",
      "withTemplate": "{{ templateValue }}",
      "withTemplateNDoubleQuotes": "{{ templateValue }}",
    },
  },
  "tests/fixtures/nested/level2/level3/page5": Object {
    "html": Object {
      "number": 123,
      "test": "Test value",
      "title": "Page5",
      "withDoubleQuotes": "Value with quotes",
      "withQuotes": "Value with quotes",
      "withQuotesInside": "Value\"with\'qoutes\'ins\"ide",
      "withSpaces": "Spaced value",
      "withTemplate": "{{ templateValue }}",
      "withTemplateNDoubleQuotes": "{{ templateValue }}",
    },
  },
  "tests/fixtures/nested/level2/page3": Object {
    "html": Object {
      "number": 123,
      "test": "Test value",
      "title": "Page3",
      "withDoubleQuotes": "Value with quotes",
      "withQuotes": "Value with quotes",
      "withQuotesInside": "Value\"with\'qoutes\'ins\"ide",
      "withSpaces": "Spaced value",
      "withTemplate": "{{ templateValue }}",
      "withTemplateNDoubleQuotes": "{{ templateValue }}",
    },
  },
  "tests/fixtures/nested/level2/pageWithoutMatter": Object {
    "html": Object {},
  },
  "tests/fixtures/nested/page1": Object {
    "html": Object {
      "number": 123,
      "test": "Test value",
      "title": "Page1",
      "withDoubleQuotes": "Value with quotes",
      "withQuotes": "Value with quotes",
      "withQuotesInside": "Value\"with\'qoutes\'ins\"ide",
      "withSpaces": "Spaced value",
      "withTemplate": "{{ templateValue }}",
      "withTemplateNDoubleQuotes": "{{ templateValue }}",
    },
  },
  "tests/fixtures/nested/page2": Object {
    "html": Object {
      "number": 123,
      "test": "Test value",
      "title": "Page2",
      "withDoubleQuotes": "Value with quotes",
      "withQuotes": "Value with quotes",
      "withQuotesInside": "Value\"with\'qoutes\'ins\"ide",
      "withSpaces": "Spaced value",
      "withTemplate": "{{ templateValue }}",
      "withTemplateNDoubleQuotes": "{{ templateValue }}",
    },
  },
}
`;

exports[`Gray Matter task should produce correct matter file with defaults for multiple pages 1`] = `
Object {
  "tests/fixtures/multiple/page1": Object {
    "html": Object {
      "number": 123,
      "test": "Test value",
      "title": "Page1",
      "withDoubleQuotes": "Value with quotes",
      "withQuotes": "Value with quotes",
      "withQuotesInside": "Value\"with\'qoutes\'ins\"ide",
      "withSpaces": "Spaced value",
      "withTemplate": "{{ templateValue }}",
      "withTemplateNDoubleQuotes": "{{ templateValue }}",
    },
  },
  "tests/fixtures/multiple/page2": Object {
    "html": Object {
      "number": 123,
      "test": "Test value",
      "title": "Page2",
      "withDoubleQuotes": "Value with quotes",
      "withQuotes": "Value with quotes",
      "withQuotesInside": "Value\"with\'qoutes\'ins\"ide",
      "withSpaces": "Spaced value",
      "withTemplate": "{{ templateValue }}",
      "withTemplateNDoubleQuotes": "{{ templateValue }}",
    },
  },
  "tests/fixtures/multiple/page3": Object {
    "html": Object {
      "number": 123,
      "test": "Test value",
      "title": "Page3",
      "withDoubleQuotes": "Value with quotes",
      "withQuotes": "Value with quotes",
      "withQuotesInside": "Value\"with\'qoutes\'ins\"ide",
      "withSpaces": "Spaced value",
      "withTemplate": "{{ templateValue }}",
      "withTemplateNDoubleQuotes": "{{ templateValue }}",
    },
  },
  "tests/fixtures/multiple/pageWithoutMatter": Object {
    "html": Object {},
  },
}
`;

exports[`Gray Matter task should produce correct matter file with defaults for single page 1`] = `
Object {
  "tests/fixtures/single/page": Object {
    "html": Object {
      "number": 123,
      "test": "Test value",
      "title": "Page",
      "withDoubleQuotes": "Value with quotes",
      "withQuotes": "Value with quotes",
      "withQuotesInside": "Value\"with\'qoutes\'ins\"ide",
      "withSpaces": "Spaced value",
      "withTemplate": "{{ templateValue }}",
      "withTemplateNDoubleQuotes": "{{ templateValue }}",
    },
  },
}
`;
